lint-rules/src/main/java/com/duckduckgo/lint/MissingSmartlingRequiredDirectivesDetector.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.lint
+
+import com.android.resources.ResourceFolderType
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Context
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Incident
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity.ERROR
+import com.android.tools.lint.detector.api.XmlContext
+import com.android.tools.lint.detector.api.XmlScanner
+import org.w3c.dom.Document
+
+@Suppress("UnstableApiUsage")
+class MissingSmartlingRequiredDirectivesDetector : Detector(), XmlScanner {
+
+    override fun appliesTo(folderType: ResourceFolderType): Boolean {
+        return folderType == ResourceFolderType.VALUES
+    }
+
+    private fun ignoreFile(context: Context) =
+        context.file.name.startsWith("donottranslate") || !context.file.name.contains("strings")
+
+    override fun visitDocument(
+        context: XmlContext,
+        document: Document
+    ) {
+        if (ignoreFile(context)) {
+            return
+        }
+
+        val contents = context.getContents() ?: return
+
+        REQUIRED_SMARTLING_DIRECTIVES.forEach {
+            if (!contents.contains(it)) {
```

Comment:
The rule is not really making sure the directives are created before resources, moving on directive anywhere in the file still passes the tests when it should not. Instead, you can iterate over the different nodes to make sure the directives exist before the resources node.

Reply:
Ahh! Good point. Completely missed that. Will update.